docs/migration.rst

```diff
@@ -0,0 +1,368 @@
+ambiguous time. To handle that situation, see `Detecting ambiguous and
+imaginary times`_.
+
+If you want "absolute time" rather than "wall time" arithmetic, the best option
+is to perform the arithmetic in UTC. Here is a simple helper function for that
+purpose (in Python 2 or 2/3 compatible code, replace ``datetime.timezone.utc``
+with ``dateutil.tz.UTC``):
+
+.. code-block:: python
+
+    from datetime import timezone
+    def absolute_add(dt, td):
+        dt_utc = dt.astimezone(timezone.utc)
+        return (dt_utc + td).astimezone(dt.tzinfo)
+
+This will have the same semantics as "add and normalize" in ``pytz``, and
+similarly guarantees that the result exists.
+
+
+Getting a time zone's name
+--------------------------
+
+``pytz`` zones have a ``.zone`` attribute that exposes the key used to created
+it from the IANA time zone database. The equivalent attribute on
+``zoneinfo.ZoneInfo`` objects is :attr:`zoneinfo.ZoneInfo.key`. There is
+currently no equivalent for this in ``dateutil`` zones.
+
+You can also recover this information by calling ``str`` on a ``pytz`` zone, a
+shim class zone (even in Python 2), or a ``zoneinfo.ZoneInfo`` zone.
+
+
+.. _constructing-tzinfo:
+
+Constructing a ``tzinfo`` object
+--------------------------------
+
+Most of this guide assumes that you already have a time zone object, because
+it is aimed at people who were using ``pytz``-specific features of a time zone
+returned by a library that is switching over to use a PEP 495-compatible time
+zone provider. However, if you are *also* creating your own ``pytz`` objects,
+or you want to switch to directly creating ``tzinfo`` objects yourself, this
+section covers creating PEP 495-compatible ``tzinfo`` objects.
+
+IANA zones
+##########
+
+With ``pytz``, one creates an IANA / Olson time zone object via the
+``pytz.timezone`` function, like so:
+
+.. code-block:: python
+
+    import pytz
+    LA = pytz.timezone("America/Los_Angeles")
+    # <DstTzInfo 'America/Los_Angeles' LMT-1 day, 16:07:00 STD>
+
+When using :mod:`zoneinfo`, instead use the :class:`zoneinfo.ZoneInfo`
+constructor. Note: in Python 3.6-3.8, replace ``import zoneinfo`` with
+``from backports import zoneinfo``:
+
+.. code-block:: python
+
+    import zoneinfo
+    LA = zoneinfo.ZoneInfo("America/Los_Angeles")
+    # zoneinfo.ZoneInfo(key='America/Los_Angeles')
+
+When using :mod:`dateutil.tz`, use :func:`dateutil.tz.gettz`:
+
+.. code-block:: python
+
+    from dateutil import tz
+    LA = tz.gettz("America/Los_Angeles")
+    # tzfile('/usr/share/zoneinfo/America/Los_Angeles')
+
+
+UTC and fixed offset zones
+##########################
+
+``pytz`` provides a convenience singleton ``pytz.UTC``, as well as a
+``FixedOffset`` function, for constructing a value with a fixed offset in
+minutes.
+
+To get an object representing UTC, in Python 3+, use the standard
+library-provided :attr:`datetime.timezone.utc` singleton. When using
+``dateutil``, use ``dateutil.tz.UTC``.
```

Comment:
given that these exist, do they really belong below the section for "Constructing a tzinfo"?

also, does it make sense to have a one-stop attribute for this that chooses for you based on py2/py3?

Reply:
> also, does it make sense to have a one-stop attribute for this that chooses for you based on py2/py3?

In the long run, I wouldn't expect end users to be *using* `pytz_deprecation_shim`, since it's more intended as a way to provide `pytz`'s interface to your end users while not actually using `pytz`, as a temporary measure.

That said, `pytz_deprecation_shim.UTC` does exist, and `pytz_deprecation_shim.helpers.upgrade_tzinfo(pytz_deprecation_shim.UTC)` would work to get "the right thing" (I'm strongly considering adding an `.unwrap` method or something to the shims in place of `helpers.upgrade_tzinfo`).

Reply:
> given that these exist, do they really belong below the section for "Constructing a tzinfo"?

I've adjusted this to be "Acquiring a `tzinfo` object"